streams/src/main/java/org/apache/kafka/streams/kstream/internals/KGroupedTableImpl.java

```diff
@@ -58,12 +59,12 @@
     KGroupedTableImpl(final InternalStreamsBuilder builder,
                       final String name,
                       final String sourceName,
-                      final Serde<K> keySerde,
-                      final Serde<V> valSerde,
+                      final GroupedInternal<K, V> groupedInternal,
                       final StreamsGraphNode streamsGraphNode) {
         super(builder, name, Collections.singleton(sourceName), streamsGraphNode);
-        this.keySerde = keySerde;
-        this.valSerde = valSerde;
+        this.keySerde = groupedInternal.keySerde();
+        this.valSerde = groupedInternal.valueSerde();
+        this.repartitionTopicName = groupedInternal.name();
```

Comment:
`repartitionTopicName` and `repartitionTopic` is a bit confusing. I'd suggest just keeping the `GroupedInternal` as a field to replace key/valueSerde and `repartitionTopicName` in the constructor and retrieve its fields later.

Ditto for other internal class's constructors (you already replaced serdes with the object in some classes, just trying to suggest consistency here).

Reply:
ack, I believe I've cleaned this up.